frontend/src/metabase/common/hooks/filters/use-number-filter/use-number-filter.ts

```diff
@@ -0,0 +1,65 @@
+import { useMemo, useState } from "react";
+import * as Lib from "metabase-lib";
+import { getAvailableOperatorOptions } from "../utils";
+import type { NumberPickerOperator } from "./types";
+import { OPERATOR_OPTIONS } from "./constants";
+import { getDefaultValues, hasValidValues, getFilterClause } from "./utils";
+
+type UseNumberFilterOpts = {
+  query: Lib.Query;
+  stageIndex: number;
+  column: Lib.ColumnMetadata;
+  filter?: Lib.FilterClause;
+  defaultOperator?: NumberPickerOperator;
+};
+
+export function useNumberFilter({
+  query,
+  stageIndex,
+  column,
+  filter,
+  defaultOperator = "=",
+}: UseNumberFilterOpts) {
+  const filterParts = useMemo(
+    () => (filter ? Lib.numberFilterParts(query, stageIndex, filter) : null),
+    [query, stageIndex, filter],
+  );
+
+  const availableOperators = useMemo(
+    () =>
+      getAvailableOperatorOptions(query, stageIndex, column, OPERATOR_OPTIONS),
+    [query, stageIndex, column],
+  );
+
+  const [operator, _setOperator] = useState(
+    filterParts ? filterParts.operator : defaultOperator,
+  );
+
+  const [values, setValues] = useState(() =>
+    getDefaultValues(operator, filterParts?.values),
+  );
+
+  const { valueCount, hasMultipleValues } = OPERATOR_OPTIONS[operator];
+  const isValid = hasValidValues(operator, values);
+
+  const setOperator = (operator: Lib.NumberFilterOperatorName) => {
```

Comment:
`handleOperatorChange` or something like that

Reply:
I know we have this convention for _components_, but do we have one for hooks? I think existing hooks mostly avoid the word "handle" in returned function names?